subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorIntegrationTest.groovy

```diff
@@ -694,6 +694,110 @@ class WorkerExecutorIntegrationTest extends AbstractWorkerExecutorIntegrationTes
+            import org.gradle.api.artifacts.ConfigurationContainer;
+            import org.gradle.api.file.FileCollection;
+            import org.gradle.api.tasks.Classpath;
+            import org.gradle.api.tasks.TaskAction;
+            import org.gradle.workers.WorkerExecutor;
+            import org.gradle.workers.WorkAction;
+            import org.gradle.workers.WorkParameters;
+            import org.gradle.workers.WorkQueue;
+
+            import javax.inject.Inject;
+
+            public class WorkerRunningPlugin implements Plugin<Project> {
+                public void apply(Project project) {
+                    ConfigurationContainer configurations = project.getConfigurations();
+                    NamedDomainObjectProvider<Configuration> bucket = configurations.register("bucket", config -> {
+                        config.setCanBeConsumed(false);
+                        config.setCanBeResolved(false);
+                    });
+
+                    NamedDomainObjectProvider<Configuration> resolvable = configurations.register("resolvable", config -> {
+                        config.setCanBeConsumed(false);
+                        config.setCanBeResolved(true);
+                        config.extendsFrom(bucket.get());
+                        config.getDependencies().add(project.getDependencies().create(project));
+                    });
+
+                    project.getTasks().register("runAction", ActionRunningTask.class, task -> {
+                        task.setClasspath(resolvable.get());
+                    });
+                }
+
+                public static abstract class ActionRunningTask extends DefaultTask {
+                    private FileCollection classpath;
+                    @Classpath
+                    public FileCollection getClasspath() { return classpath; }
+                    public void setClasspath(FileCollection classpath) { this.classpath = classpath; }
+
+                    @Inject
+                    public abstract WorkerExecutor getWorkerExecutor();
+
+                    @TaskAction
+                    public void execute() {
+                        WorkQueue workQueue = getWorkerExecutor().processIsolation(spec -> {
+                            spec.getClasspath().from(classpath);
+                        });
+                        workQueue.submit(TestAction.class, x -> {});
+                    }
+                }
+
+                public static abstract class TestAction implements WorkAction<WorkParameters.None> {
+                    @Override
+                    public void execute() {
+                        // Make the ClassLoader "dirty" by trying to load something from it
+                        Thread.currentThread().getContextClassLoader().getResource("abc");
+                    }
+                }
+            }
+        """
+
+        when:
+        succeeds("runAction", "--info")
+
+        then:
+        outputContains("Started Gradle worker daemon")
```

Comment:
This might be overkill as long as we document that we have to use process isolation in the test, that should be good enough.